util.go

```diff
@@ -19,6 +19,14 @@ func renameWithFallback(src, dest string) error {
 		return err
 	}
 
+	// Windows cannot use syscall.Rename to rename a directory
```

Comment:
OK forgive me for being obtuse about this, but I'm trying to understand the background on this limitation, and I've yet to turn up anything super clear. From what I can gather, Go [relies on the MoveFile API](https://github.com/golang/go/issues/6887#issuecomment-66088858) in general; for [`syscall.Rename()`](https://golang.org/src/syscall/syscall_windows.go#L423), it calls [`MoveFile()`](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365239(v=vs.85).aspx), which I guess should be capable of handling dir renames.

[`os.Rename`](https://golang.org/src/os/file.go?s=8019:8061#L259), [otoh](), also [ultimately uses](https://golang.org/src/internal/syscall/windows/syscall_windows.go#L130) the MoveFile API to do its work - which means it should also be capable of dir renames? The only difference is that it uses [`MoveFileEx`](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365245(v=vs.85).aspx), with the `MOVEFILE_REPLACE_EXISTING` flag (I believe that's maybe new as of Go 1.6?), which makes behavior generally more similar to POSIX systems than a pure `MoveFile()` call.

So yeah, I'm confused on this, and pointers to some explanation of the limitation here would be super appreciated.